src/sql/platform/connection/test/common/connectionConfig.test.ts

```diff
@@ -603,7 +603,7 @@ suite('ConnectionConfig', () => {
 			providerName: 'MSSQL',
 			options: {},
 			saveProfile: true,
-			id: 'server3',
+			id: 'server3-2',
```

Comment:
I'm not sure this is correct? The ids should be different but currently your update changes them to both be the same.

What was the error you were seeing?

Reply:
Well, the tests were failing since this wasn't throwing.

Before, instead of comparing the IDs we were comparing the getConnectionInfoId(), which is somewhat different.

Hmm. I'll see if I can do this without converting to a ConnectionProfile